src/goto-symex/symex_assign.cpp

```diff
@@ -434,16 +434,10 @@ void goto_symext::symex_assign_symbol(
     return;
   }
 
-  exprt l2_rhs = [&]() {
-    exprt guarded_rhs = rhs;
+  exprt l2_rhs = state.rename(
     // put assignment guard into the rhs
-    if(!guard.empty())
-    {
-      guarded_rhs =
-        if_exprt{conjunction(guard), std::move(guarded_rhs), lhs, rhs.type()};
-    }
-    return state.rename(std::move(guarded_rhs), ns).get();
-  }();
+    guard.empty() ? rhs : (exprt)if_exprt{conjunction(guard), rhs, lhs}, ns)
```

Comment:
I'd prefer `static_cast<exprt>` as the C-style cast is just very easy to overlook.